### DISCUSSION TITLE
Why missing secrets are no longer asked by the deploy UI

### DISCUSSION THREAD

**Discussion by mathix420** [vercel/vercel#4936] [Help] [answered]
I have a `vercel.json` file which looks like this, this file was working fine to deploy [my project](https://github.com/mathix420/notion-charts) via a [Github Button](https://vercel.com/docs/v2/more/deploy-button).

```json
{
    ...
    "build": {
        "env": {
            "TOKEN_V2": "@token_v2"
        }
    },
    "env": {
        "TOKEN_V2": "@token_v2"
    }
}
```

To make my project usable for non-CLI users I have no choice to convert secrets into env variables. But these are less convenient as I have 2 projects using the same token.

If anyone has any idea to avoid using environment variables I would be happy to hear about it.

**Comment by styfle** [comment, marked as answer]
Take a look at the Deploy Button generator to see how you can add environment variables you expect for users cloning/forking your project:

https://vercel.com/docs/v2/more/deploy-button

We might add support for global secrets in the UI, but it is not supported at this time.
All environment variables are defined per project.

**Reply by mathix420** [reply]
That's what I've used but the point is that `environment variables are defined per project`, as you said.

+1 for global secrets in the UI hope this will be integrated one day!

Thanks for the response 😉
